### PR TITLE
adding qr ics code

### PR DESCRIPTION
**Pull request by teiling88**
### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code

### CODE REVIEW

File: htdocs/src/Oc/GeoCache/Controller/GeoCacheFileController.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    /**
+     * @param Request $request
+     * @Route("/api/geocache/qrCodes/ics")
+     */
+    public function generateQrCodeIcs(Request $request): void
+    {
+        $waypoint = $request->get('wp');
+        $geoCache = $this->geoCacheService->fetchByWaypoint($waypoint);
+
+        if (!$geoCache instanceof GeoCacheEntity && $geoCache->type !== 6) {
```
Comment (Crease29) [inline htdocs/src/Oc/GeoCache/Controller/GeoCacheFileController.php:61]: What means 6? Can we use public constants or static properties for that?

File: htdocs/src/Oc/GeoCache/Util.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+use Endroid\QrCode\ErrorCorrectionLevel;
+use Endroid\QrCode\LabelAlignment;
+use Endroid\QrCode\QrCode;
+use Oc\GeoCache\Persistence\GeoCache\GeoCacheEntity;
+
+class Util
+{
+    public function generateIcsStringFromGeoCache(GeoCacheEntity $geoCache): string
+    {
+        if ($geoCache->type !== 6) {
```
Comment (Crease29) [inline htdocs/src/Oc/GeoCache/Util.php:20]: Also here: What means 6?

File: htdocs/templates2/ocstyle/viewcache.tpl [hunk 3]
```diff
@@ -306,7 +306,8 @@
                     <option value="search.php?searchto=searchbycacheid&showresult=1&f_inactive=0&f_ignored=0&startat=0&cacheid={$cache.cacheid}&output=ovl">OVL</option>
                     <option value="search.php?searchto=searchbycacheid&showresult=1&f_inactive=0&f_ignored=0&startat=0&cacheid={$cache.cacheid}&output=txt">TXT</option>
                     {if $cache.type==6}
```
Comment (Crease29) [inline htdocs/templates2/ocstyle/viewcache.tpl:308]: And here.

File: htdocs/templates2/ocstyle/viewcache.tpl [hunk 4]
```diff
@@ -543,7 +544,12 @@ function showalllists()
             <a href="search.php?searchto=searchbycacheid&showresult=1&f_inactive=0&f_ignored=0&startat=0&cacheid={$cache.cacheid}&output=ov2" rel="nofollow" title="{t}TomTom POI .ov2{/t}">OV2</a> -
             <a href="search.php?searchto=searchbycacheid&showresult=1&f_inactive=0&f_ignored=0&startat=0&cacheid={$cache.cacheid}&output=ovl" rel="nofollow" title="{t}TOP50-Overlay .ovl{/t}">OVL</a> -
             <a href="search.php?searchto=searchbycacheid&showresult=1&f_inactive=0&f_ignored=0&startat=0s&cacheid={$cache.cacheid}&output=txt" rel="nofollow" title="{t}Textfile .txt{/t}">TXT</a>
-            <br />
+            {if $cache.type==6}
```
Comment (Crease29) [inline htdocs/templates2/ocstyle/viewcache.tpl:547]: And here.

File: htdocs/src/Oc/GeoCache/Controller/GeoCacheFileController.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            throw new \InvalidArgumentException('the waypoint is not valid!');
+        }
+
+        header('Content-Type: image/png');
+
+        if ($request->get('download')) {
+            header('Content-Disposition: attachment; filename="' . $waypoint . '.png"');
+        }
+
+        $this->geoCacheUtil->generateQrCodeFromString('https://www.opencaching.de/' . $geoCache->wpOc);
```
Comment (nlubisch) [inline htdocs/src/Oc/GeoCache/Controller/GeoCacheFileController.php:49]: > https://www.opencaching.de/

Can you refactor this be a global parameter?

File: htdocs/src/Oc/GeoCache/Controller/GeoCacheFileController.php [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        }
+
+        $icsString = $this->geoCacheUtil->generateIcsStringFromGeoCache($geoCache);
+
+        if ($request->get('download')) {
+            header('Content-Type: text/calendar; charset=utf-8');
+            header('Content-Disposition: attachment; filename="' . $geoCache->wpOc . '.ics"');
+
+            echo $icsString;
+            die();
```
Comment (nlubisch) [inline htdocs/src/Oc/GeoCache/Controller/GeoCacheFileController.php:72]: die :(

File: htdocs/src/Oc/GeoCache/Util.php [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+            throw new \InvalidArgumentException('the given geoCache is not an event cache!');
+        }
+
+        $vCalendar = new Calendar('https://www.opencaching.de/' . $geoCache->wpOc);
```
Comment (nlubisch) [inline htdocs/src/Oc/GeoCache/Util.php:24]: Can you refactor this be a global parameter?

File: htdocs/src/Oc/GeoCache/Util.php [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        $vEvent = new Event();
+
+        $vEvent
+            ->setDtStart($geoCache->dateHidden)
+            ->setDtEnd($geoCache->dateHidden->add(new \DateInterval('PT1H')))
+            ->setNoTime(true)
+            ->setSummary($geoCache->name)
+            ->setDescription('https://www.opencaching.de/viewcache.php?cacheid=' . $geoCache->cacheId);
```
Comment (nlubisch) [inline htdocs/src/Oc/GeoCache/Util.php:32]: Can you refactor this be a global parameter?

**Pull request merged**